src/platform_specific/named_threads.win.cc

```diff
@@ -0,0 +1,69 @@
+#include "named_threads.h"
+
+#include <cassert>
+#include <cwchar>
+#include <type_traits>
+
+#include <windows.h>
+
+namespace celerity::detail {
+
+static_assert(std::is_same_v<std::thread::native_handle_type, HANDLE>, "Unexpected native thread handle type");
+
+template <typename SrcT, typename DstT>
+static inline DstT convert_string(const SrcT& str) {
+	static_assert(
+	    (std::is_same_v<SrcT, std::string> && std::is_same_v<DstT, std::wstring>) || (std::is_same_v<SrcT, std::wstring> && std::is_same_v<DstT, std::string>),
+	    "Unsupported string type");
+
+	constexpr auto converter = [](typename DstT::value_type* dst, const typename SrcT::value_type** src, const std::size_t len, std::mbstate_t* ps) {
```

Comment:
It looks like code de-duplication has gone a bit off the rails here. Wouldn't two overloads
```c++
std::string convert_string(const std::wstring&);
std::wstring convert_string(const std::string&);
```
with mostly identical implementations (up to `mbsrtowcs` vs `wcsrtombs`) have fewer total lines of code?

Reply:
This is actually what I had originally. I set out to de-duplicate this thinking it would be a one-liner to select between `mbsrtowcs`/`wcsrtombs`, which turned out not to be the case. So now I reverted it to what I had before de-duplicating.